plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterPrice.php

```diff
@@ -120,144 +120,73 @@ protected function render( $attributes, $content, $block ) {
-			'data-has-filter'     => 'no',
+			'data-wc-context'      => wp_json_encode(
+				array(
+					'minPrice'           => $min_price,
+					'maxPrice'           => $max_price,
+					'minRange'           => $min_range,
+					'maxRange'           => $max_range,
+					'hasFilterOptions'   => $min_range !== $max_range,
+					'hasSelectedFilters' => $min_price !== $min_range || $max_price !== $max_range,
+				),
+				JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP,
+			),
+			'data-wc-key'          => 'product-filter-price-' . md5( wp_json_encode( $attributes ) ),
+			'data-wc-bind--hidden' => '!context.hasFilterOptions',
 		);
 
 		if ( $min_range === $max_range || ! $max_range ) {
 			return sprintf(
-				'<div %s></div>',
-				get_block_wrapper_attributes( $wrapper_attributes )
+				'<div %1$s hidden>%2$s</div>',
+				get_block_wrapper_attributes( $wrapper_attributes ),
+				array_reduce(
+					$block->parsed_block['innerBlocks'],
+					function ( $carry, $parsed_block ) {
+						$carry .= render_block( $parsed_block );
+						return $carry;
+					},
+					''
+				)
 			);
 		}
 
-		// CSS variables for the range bar style.
-		$__low       = 100 * ( $min_price - $min_range ) / ( $max_range - $min_range );
-		$__high      = 100 * ( $max_price - $min_range ) / ( $max_range - $min_range );
-		$range_style = "--low: $__low%; --high: $__high%";
-
-		$price_min = $show_input_fields ?
-			sprintf(
-				'<input
-					class="min"
-					name="min"
-					type="text"
-					value="%s"
-					data-wc-bind--value="state.formattedMinPrice"
-					data-wc-on--input="actions.updateProducts"
-					data-wc-on--focus="actions.selectInputContent"
-					pattern=""
-				/>',
-				wp_strip_all_tags( $formatted_min_price )
-			) : sprintf(
-				'<span data-wc-text="state.formattedMinPrice">%s</span>',
-				// Not escaped, as this is HTML.
-				$formatted_min_price
-			);
-
-		$price_max = $show_input_fields ?
-			sprintf(
-				'<input
-					class="max"
-					name="max"
-					type="text"
-					value="%s"
-					data-wc-bind--value="state.formattedMaxPrice"
-					data-wc-on--input="actions.updateProducts"
-					data-wc-on--focus="actions.selectInputContent"
-				/>',
-				wp_strip_all_tags( $formatted_max_price )
-			) : sprintf(
-				'<span data-wc-text="state.formattedMaxPrice">%s</span>',
-				// Not escaped, as this is HTML.
-				$formatted_max_price
-			);
-
-		$wrapper_attributes['data-has-filter'] = 'yes';
-
-		$filter_price_content_classes = array(
-			'wp-block-woocommerce-product-filter-price-content',
-			$show_input_fields && $inline_input ? 'wp-block-woocommerce-product-filter-price-content--inline' : '',
+		return sprintf(
+			'<div %1$s>%2$s</div>',
+			get_block_wrapper_attributes( $wrapper_attributes ),
+			array_reduce(
+				$block->parsed_block['innerBlocks'],
+				function ( $carry, $parsed_block ) use ( $filter_context ) {
+					$carry .= ( new \WP_Block( $parsed_block, array( 'filterData' => $filter_context ) ) )->render();
```

Comment:
I'm working on #51638 and noticed this code. While I overall understand the needed, I'm not sure that we should use `BlockContextProvider` in block context. [Checking the docs](https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/block-context/README.md), it says:

> The reason BlockContext is only internally available within the @wordpress/block-editor module is to reinforce the expectation that external consumption of values from block context should be declared on the block registration using the context property.


Instead, should we try to use the [BlockContext approach](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/)?

This approach looks like the standard method for passing data across blocks. Additionally, it eliminates the need to implement this logic manually, as the context will be automatically defined in ProductFilterChips on the PHP side.

Reply:
As discussed in our chat, we're using only the public interfaces (`BlockContextProvider` and `WP_Block`) to pass the context from the parent to the inner blocks. We did this way because of the limitation of block context, which is 1:1 mapping to the block attribute. The filter data is dynamic so it can't be a block attribute. In addition, we render blocks here like how `render_block` does under the hood.